File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefIntegerProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefIntegerProperty extends CTPrefProperty<Integer> implements CTNumberProperty<Integer> {
    public CTPrefIntegerProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, 0);
    }

    public CTPrefIntegerProperty(@NotNull CTPreferencesProvider provider, String name, int def) {
        super(Integer.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefIntegerProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefIntegerProperty @JvmOverloads constructor(provider: CTPreferencesProvider, name: String, def: Int = 0) :
    CTPrefProperty<Int>(
        Int::class, provider, name, def
    ), CTNumberProperty<Int>
